### PR TITLE
Migrations

### PR DESCRIPTION
**Pull request by rcky**
Migrations prototype, this is really rudimentary and transactions do not work yet.

https://github.com/saia-burgess-controls/website-issues/issues/480

This branch is based on the initial setup. We'll need to clean all up.

### CODE REVIEW

File: app/config/datasources.test.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "db": {
+    "name": "db",
+    "connector": "postgresql",
+    "host": "0.0.0.0",
+    "port": 5432,
+    "database": "microservice_test",
+    "user":"microservice_test",
+    "password": "test"
```
Comment (lkappeler) [inline app/config/datasources.test.json]: do we have a possibility to not store the passwords on github?

File: lib/Microservice.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        });
+
+    }
+
+    /**
+     * Boots the application.
+     *
+     * More information on the options can be found here https://apidocs.strongloop.com/loopback-boot/.
+     *
+     * @todo: Do we need to protect the app from being booted multiple times?
```
Comment (lkappeler) [inline lib/Microservice.js]: pleas dont use todos in pull requests see https://github.com/joinbox/guidelines/blob/master/styleguide/javascript.md 18.4

File: lib/MigrationTask.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
+const Task = require('./Task');
+
+module.exports = class MigrationTask extends Task {
+
+    constructor(task, rerun = false) {
```
Comment (lkappeler) [inline lib/MigrationTask.js]: can u add some comments to the functions and there documenting the parameter eg the taks parameter needs to be an instance of tasks

File: app/tasks/UpdateModels.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+module.exports = class UpdateModels extends Task {
+
+    constructor() {
+        super('Automigrate-Models', '1.0.0');
+    }
+
+    run(dependencies, result) {
+        const datasources = dependencies.service.app.datasources;
+        const sources = this.sourcesToUniqueArray(datasources);
```
Comment (lkappeler) [inline app/tasks/UpdateModels.js]: What the hack do you get from loopback as datasources so this is necessary?

**Comment by eventEmitter** [comment]
@rcky any progress?

**Comment by rcky** [comment]
Will be adjusted to use the new migrations package.

**Pull request closed without merging**